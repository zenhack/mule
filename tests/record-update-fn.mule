#XFAIL: type formatting
fn x. x where { y = {} }
