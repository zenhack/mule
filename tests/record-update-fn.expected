
Desugared:

	(fn x (((.= y) x) {}))

inferred type:

	(all a
	 (all b
	  ((record (types (...b)) (values (...a))) ->
	   (record (types (...b))
	    (values ((y (record (types ()) (values ()))) ...a))))))

Runtime term:

	(Lam
	 (0 ()
	  (App
	   ((App
	     ((Lam
	       (0 () (Lam (1 () (Update (old (Var 1)) (label y) (field (Var 0)))))))
	      (Var 0)))
	    (Record ())))))

Evaluated:

	(Lam
	 (0 ()
	  (App
	   ((App
	     ((Lam
	       (0 () (Lam (1 () (Update (old (Var 1)) (label y) (field (Var 0)))))))
	      (Var 0)))
	    (Record ())))))
